nostr: helpers to extract `nostr:` URIs from text

yukibtc opened the pull request:
- Add `nip21::extract_from_text`
- Add `Tags::from_text`

Closes https://github.com/rust-nostr/nostr/issues/751

yukibtc commented:
@dluvian

dluvian commented on crates/nostr/src/nips/nip21.rs:
```diff
@@ -179,10 +184,67 @@ impl Nip21 {
     }
 }
 
+/// Extract `nostr:` URIs from a text
```
Since I want to use this to add tags to my events, it's helpful to know that I should deduplicate the result vector myself

```suggestion
/// Extract `nostr:` URIs from a text
///
/// The returned vector maintains the same order as their occurrences in the `input`. 
/// The result is not deduplicated, meaning that if a URI appears multiple times in the input, 
/// it will appear the same number of times in the output.
```

dluvian reviewed (approved):
LGTM

yukibtc commented:
@dluvian, before continue with this PR, I've opened #755.

In that PR I've added `Tags::dedup` for tags deduplication, which is required for the `Tags::from_text` that I've added here.
When #755 will be completed, we will go ahead with this

dluvian reviewed (approved):
LGTM
